### DISCUSSION TITLE
Ошибки в адмике на PHP8.1

### DISCUSSION THREAD

**Discussion by kelemator** [maxsite/cms#497] [Q&A]
Здравствуйте. Стараюсь периодически обновлять движок до последней текущей версии. При последнем обновлении вылезли ошибки в админке:

1) на странице "Мой профиль":
A PHP Error was encountered
Severity: 8192
Message: htmlspecialchars(): Passing null to parameter #1 ($string) of type string is deprecated
Filename: admin_users/my_profile.php
Line Number: 153
-
A PHP Error was encountered
Severity: 8192
Message: htmlspecialchars(): Passing null to parameter #1 ($string) of type string is deprecated
Filename: admin_users/my_profile.php
Line Number: 156

2) на странице "Комментаторы" (в 2-х разных местах):
A PHP Error was encountered
Severity: 8192
Message: Automatic conversion of false to array is deprecated
Filename: admin/common.php
Line Number: 212

Полагаю, это требует каких-то дополнительных правок, чтобы соответствовало PHP8.1? Или я не прав и это у меня какие-то косяки?

Спасибо за ответ :)

**Comment by maxsite** [comment]
Гляну немного позже.

**Reply by kelemator** [reply]
Здравствуйте! Планируется ли обновление движка? Я так понимаю работы приостановлены из-за войны. 
Возможно, мой вопрос не очень уместен, но все-таки, интересно какие у вас планы на будущее. Будет очень грустно, если развитие платформы на этом остановится.
Заранее спасибо за ответ.

**Reply by maxsite** [reply]
Вы правильно понимаете. Пока война, никаких работ не будет. После уже будем думать. Если доживём...

**Reply by kelemator** [reply]
Пожелаю вам силы и мужества, чтобы пережить этот ужас. Пусть у вас все будет хорошо!

**Comment by Ivanych2302** [comment]
Проблема:
При вызове функции `htmlspecialchars()` в PHP возникает ошибка `Passing null to parameter #1 ($string) of type string is deprecated`.

Решение:
Чтобы устранить ошибку, нужно убедиться, что значение, передаваемое в функцию `htmlspecialchars()`, не является `null`. Можно добавить проверку перед вызовом функции и заменить `null` на пустую строку. Вот пример кода:

```php
$admin_note = $row['users_admin_note'] ?? '';

$sanitized_admin_note = htmlspecialchars($admin_note, ENT_QUOTES, 'UTF-8');

$CI->table->add_row(t('Примечание админа'), '<textarea name="f_admin_note" cols="90" rows="3">' . $sanitized_admin_note . '</textarea>');

**Reply by maxsite** [reply]
Какой файл?

**Comment by Ivanych2302** [comment]
my_profile.php   стр 153/ 156                  первый раз поправил и потом переустанавливал несколько раз на эту базу проблем нет. Сейчас установил на другую бд опять проблема .Вне.с код проблема исчезла. PHP 8.2

**Comment by maxsite** [comment]
Попробуйте такой вариант, а то я не могу воспоризвести проблему у себя.

```
$CI->table->add_row(t('Описание'), '<textarea name="f_description" cols="90" rows="3">' . htmlspecialchars($row['users_description'] ?? '') . '</textarea>');
				
if ( mso_check_allow('edit_users_admin_note') )
    $CI->table->add_row(t('Примечание админа'), '<textarea name="f_admin_note" cols="90" rows="3">' . htmlspecialchars($row['users_admin_note'] ?? '') . '</textarea>'); 
```

**Reply by Ivanych2302** [reply]
Проверил работает

**Reply by Ivanych2302** [reply]
изменил код установил проблем нет
